pensieve/cli.py

```diff
@@ -41,27 +47,75 @@ def convert(self, value, param, ctx):
 )
 @click.option("--dataset_id", "--dataset-id", default="mozanalysis", help="Dataset to write to")
 @click.option(
-    "--date",
+    "--start_date",
+    "--start-date",
+    type=ClickDate(),
+    help="First date for which data should be analyzed",
+    metavar="YYYY-MM-DD",
+)
+@click.option(
+    "--end_date",
+    "--end-date",
     type=ClickDate(),
     help="Last date for which data should be analyzed",
     metavar="YYYY-MM-DD",
 )
+@click.option(
+    "--experiment_slug",
+    "--experiment-slug",
+    help="Normandy slug of the experiment to rerun analysis for",
+)
 @click.option("--dry_run/--no_dry_run", help="Don't publish any changes to BigQuery")
-def run(project_id, dataset_id, date, dry_run):
+def run(project_id, dataset_id, start_date, end_date, experiment_slug, dry_run):
     """Fetches experiments from Experimenter and runs analysis on active experiments."""
     # fetch experiments that are still active
     collection = ExperimentCollection.from_experimenter()
-    if date is None:
-        date = format_date(datetime.today())
+    if start_date is None:
+        start_date = format_date(datetime.today())
+    else:
+        start_date = format_date(start_date)
+
+    if end_date is None:
+        end_date = format_date(datetime.today())
     else:
-        date = format_date(date)
+        end_date = format_date(end_date)
 
-    active_experiments = collection.end_on_or_after(date).of_type(("pref", "addon"))
+    active_experiments = collection.end_on_or_after(start_date).of_type(("pref", "addon"))
 
-    # Create a trivial configuration containing defaults
+    if experiment_slug is not None:
+        # run analysis for specific experiment
+        active_experiments = active_experiments.with_slug(experiment_slug)
```

Comment:
In the future, if there is a need, we could also support string patterns with placeholders for slugs.